Synview-backend/routes/user_routes.ts

```diff
@@ -0,0 +1,170 @@
+type AppState = {
+  session: Session;
+};
+const router = new Router<AppState>();
+const prisma = new PrismaClient().$extends(withAccelerate());
+
+export const EmailRegisterRequestSchema = z.object({
+  ["username"]: z.string(),
+  ["email"]: z.string().email(),
+  ["password"]: z.string(),
+});
+export const EmailLoginRequestSchema = z.object({
+  ["email"]: z.string().email(),
+  ["password"]: z.string(),
+});
+
+///
+
+// router.use(
+//   oakCors({
+//     origin: "*",
+//     methods: ["POST", "PUT", "DELETE", "GET"],
+//     allowedHeaders: ["Content-type", "Authorization"],
+//   })
+// );
+// unprotectedRouter.use(
+//   oakCors({
+//     origin: "*",
+//     methods: ["POST", "PUT", "DELETE", "GET"],
+//     allowedHeaders: ["Content-type", "Authorization"],
+//   })
+// );
+
+router
+  .post("/register", async (context) => {
+    const body = await context.request.body.json();
+    const parsedBody = EmailRegisterRequestSchema.parse(body);
+
+    const { username, email, password } = parsedBody;
+    try {
+      if (!username || !email || !password) {
+        context.response.status = 400;
+        context.response.body = {
+          error: "Recieved invalid request",
+        };
+        return;
+      }
+
+      const existingUser = await prisma.user.findFirst({
+        where: {
+          username: String(username),
+        },
+      });
+
+      const existingEmail = await prisma.user.findFirst({
+        where: {
+          email: String(email),
+        },
+      });
+
+      if (existingUser || existingEmail) {
+        context.response.status = 400;
+        context.response.body = {
+          error: "Username or Email, already taken",
+        };
+        return;
+      }
+    } catch (error) {
+      context.response.status = 400;
+      context.response.body = {
+        error: error,
+      };
+      return;
+    }
+
+    const hashedPassword = await hash(password);
+
+    await prisma.user.create({
+      data: { username: username, email: email, passwordHash: hashedPassword },
+    });
+    context.response.body = "New user created!";
+  })
+  .post("/login", async (context) => {
+    const body = await context.request.body.json();
+    try {
+      const parsedBody = EmailLoginRequestSchema.parse(body);
+      const { password, email } = parsedBody;
+
+      if (!email || !password) {
+        context.response.status = 400;
+        context.response.body = {
+          error: "Recieved invalid login request",
+        };
+        return;
+      }
+
+      const user = await prisma.user.findFirst({
+        where: {
+          email: String(email),
+        },
+      });
+
+      if (!user) {
+        context.response.status = 400;
+        context.response.body = {
+          error: "Email doesn't exist",
+        };
+        return;
+      }
+
+      const isValidPassowrd = await bycryptVerify(password, user.passwordHash);
+
+      if (!isValidPassowrd) {
+        context.response.status = 400;
```

Comment:
I would recommend creating an enum with the different HTTP codes rather than just hardcoding 400 for example.